cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -228,6 +247,26 @@ public void processMessage(MetadataMessage message) {
     }
   }
 
+  /**
+   * The {@link MetadataMessageProcessor} for processing field lineage.
+   */
+  private final class FieldLineageProcessor implements MetadataMessageProcessor {
+
+    private final FieldLineageDataset fieldLineageDataset;
+
+    FieldLineageProcessor(DatasetContext datasetContext) {
+      this.fieldLineageDataset = FieldLineageDataset.getFieldLineageDataset(datasetContext, datasetFramework,
+                                                                            fieldLineageDatasetId);
+    }
+
+    @Override
+    public void processMessage(MetadataMessage message) {
+      FieldLineageInfo info = message.getPayload(GSON, FieldLineageInfo.class);
+      ProgramRunId programRunId = message.getProgramId().run(message.getRunId());
+      fieldLineageDataset.addFieldLineageInfo(programRunId, info);
```

Comment:
Based on the current logic, if message cannot be processed, transaction fails. For messages belonging to the metadata topic, we do have retry strategy of retrying for 1000 times by default(`system.metadata.retry.policy.max.retries`). However after retrying for 1000 times, I don't see we update the messageid for ignoring the messages that are causing failures.
@chtyim is my understanding correct?

Reply:
so it will never get past the failure?

Reply:
Currently it won't skip if there is exception raised from the `processMessage`. It is expected the child class (like this one) to catch exceptions that it consider skippable and skip the message (e.g. if the MetadataMessage is considered malformed and can never be processed).

BTW, @sagarkapare I think we should also update the logic in the `AbstractMessagingSubscriberService.runOneIteration()` method, as currently it will do a no-sleeping retry after the `RetryStrategy` exhausted (meaning returning 0). We should probably default to some non-zero sleep time when that happen. I don't want to terminate the subscriber service since I don't think one subscriber bug should cause all subscribers to halt.

Reply:
@chtyim Filed this JIRA - https://issues.cask.co/browse/CDAP-13594.. will need some discussion to better understand the problem.